lib/fizzy/execute.hpp

```diff
@@ -19,17 +20,28 @@ struct Instance;
 
 using ImportedFunction = execution_result (*)(Instance&, std::vector<uint64_t>);
 
+struct ImportedMemory
+{
+    bytes* data = nullptr;
+    Limits limits;
+};
+
 struct ImportedGlobal
 {
     uint64_t* value = nullptr;
     bool is_mutable = false;
 };
 
+using bytes_ptr = std::unique_ptr<bytes, void (*)(bytes*)>;
+
 // The module instance.
 struct Instance
 {
     const Module& module;
-    bytes memory;
+    // Memory is either allocated and owned by the instance or imported as already allocated bytes
+    // and owned externally.
+    // For these cases unique_ptr would either have a normal deleter or noop deleter respectively
+    bytes_ptr memory;
```

Comment:
I guess when we're going to have a real external API, this will need to be replaced with something more complicated like an interface allowing to access and grow external memory.